src/app/debug/table/table.component.ts

```diff
@@ -662,4 +667,32 @@ export class TableComponent implements OnInit, OnDestroy {
     }
     return result;
   }
+
+  getMetadata(report: Report, field: string): string {
+    return report[field as keyof Report];
+  }
+
+  sortData(sort: Sort): any {
+    if (!sort.active || sort.direction === '') {
+      this.sortedReportMetadata = [...this.tableSettings.reportMetadata];
+      return;
+    }
+    this.sortedReportMetadata.sort((a: Report, b: Report): number => {
+      const isAsc: boolean = sort.direction === 'asc';
+      const headersA: [string, string][] = Object.entries(a);
+      const headersB: [string, string][] = Object.entries(b);
+      for (const [index, element] of headersA.entries()) {
+        if (Number(sort.active) === index) {
+          return compare(element[1], headersB[index][1], isAsc);
+        }
+      }
+      return 0;
+    });
+  }
+
+  protected readonly String = String;
```

Comment:
?

Reply:
Removed this line.
My IDE added this when I tried converting index in the template to string. I now changed it so the sort works based on the header.
This did mean I had to add another method which converted the metadata label header to the metadataname field of the report type.